standards/api_security_standards.md

```diff
@@ -0,0 +1,158 @@
+
+5.4 Endpoint Security Review: Conduct regular reviews of all API endpoints to identify and rectify any function level authorization flaws, taking into account the application's business logic and user hierarchy.
+
+5.5 Role Hierarchy and Endpoint Mapping: Clearly define and maintain a mapping of which roles have access to specific functions and ensure that all administrative functions are securely gated behind proper authorization checks.
+
+**6. Mass Assignment Prevention**
+
+Binding client provided data (e.g., JSON) to data models, without proper properties filtering based on a whitelist, usually lead to Mass Assignment. Either guessing objects properties, exploring other API endpoints, reading the documentation, or providing additional object properties in request payloads, allows attackers to modify object properties they are not supposed to.
+
+6.1 Property Whitelisting: Implement whitelisting to strictly control which properties clients are allowed to modify via API requests. This prevents clients from updating sensitive or unintended properties.
+
+6.2 Avoid Automatic Binding: Minimize the use of frameworks and functions that automatically bind client inputs to internal object properties, to mitigate unauthorized updates.
+
+6.3 Sensitive Property Blacklisting: Utilize built-in features or custom logic to blacklist sensitive or non-editable properties, ensuring they are not exposed or modifiable via API endpoints.
+
+6.4 Schema Enforcement: Define and enforce strict schemas for input data payloads to ensure only allowed and intended data is processed by the API.
+
+**7. Security Misconfiguration**
+
+Security misconfiguration is commonly a result of unsecure default configurations, incomplete or ad-hoc configurations, open cloud storage, misconfigured HTTP headers, unnecessary HTTP methods, permissive Cross-Origin resource sharing (CORS), and verbose error messages containing sensitive information.
+
+7.1 System Hardening and Configuration Reviews: Regularly review and harden configurations across the entire API stack, including network services, application layers, and cloud configurations, to eliminate unnecessary services and close security gaps.
+
+7.2 Patch Management: Ensure that all components of the API stack are up-to-date with the latest security patches to protect against vulnerabilities.
+
+7.3 Disable Unused Features: Turn off unnecessary features and HTTP verbs that are not required by the API to minimize the attack surface.
+
+7.4 Secure Communication Protocols: Mandate the use of secure communication protocols such as TLS for all API interactions, including static asset transfers, to protect data in transit.
+
+7.5 Security Headers and CORS Policy: Implement essential security headers and a properly configured Cross-Origin Resource Sharing (CORS) policy to enhance security in client-side interactions.
+
```

Comment:
Let's add example snippet for this?

Reply:
As discussed, the security standards are generic and can be applied to any programming language.

Resolution: We won't be adding snippets